### PR TITLE
Adaptive Envs, Bezier Curves, Room Dims 

### PR DESCRIPTION
**Pull request by SumeetBatra**
- Bezier curves are now randomly sampled in a while loop until a new goal pos that meets the constraints is found 
- Changed how room dims are defined and how room bounds checking works to be consistent across all files 
- Moved the floor of the room_box mesh to be slightly lower than the actual floor to prevent visual glitching 
- Purely random changing of the room dims during training and test time now supported

### CODE REVIEW

File: gym_art/quadrotor_multi/quadrotor_multi.py [hunk 1]
```diff
@@ -178,7 +178,8 @@ def reset(self):
         if self.adaptive_env:
             # TODO: introduce logic to choose the new room dims i.e. based on statistics from last N episodes, etc
             # e.g. self.room_dims = ....
-            pass
+            new_length, new_width, new_height = np.random.randint(30, 30, 30)
```
Comment (alex-petrenko) [inline gym_art/quadrotor_multi/quadrotor_multi.py]: How does this work? I've never seen randint used like this
https://numpy.org/doc/stable/reference/random/generated/numpy.random.randint.html 
Docs say the first 3 arguments are low, high and size. So what does 30, 30, 30 do here?
I'm just curious how this works.

File: gym_art/quadrotor_multi/quadrotor_multi.py [hunk 2]
```diff
@@ -268,6 +269,13 @@ def step(self, actions):
         self.all_collisions = {'drone': np.sum(drone_col_matrix, axis=1), 'ground': ground_collisions,
                                'obstacle': col_obst_quad.sum(axis=0)}
 
+        # compute clipped 1/x^2 cost for distance b/w drones
+        dists = spatial.distance_matrix(x=self.pos, y=self.pos)
+        dt = 1.0 / self.envs[0].control_freq
+        spacing_reward = hyperbolic_proximity_penalty(dists, dt)
```
Comment (alex-petrenko) [inline gym_art/quadrotor_multi/quadrotor_multi.py:293]: I think this ought to have the coefficient if we're committing this to main branch.
Something similar to self.rew_coeff["quadcol_bin"] and it should be zero by default, at least until we find a good value

**Review by alex-petrenko — CHANGES_REQUESTED** [review]

File: gym_art/quadrotor_multi/quadrotor_multi.py (hunk 1 above)
Comment (SumeetBatra) [reply gym_art/quadrotor_multi/quadrotor_multi.py]: Oof, I must've been drunk or something when I wrote that. Should be `randint(1,31,3)`

File: gym_art/quadrotor_multi/quadrotor_multi.py (hunk 2 above)
Comment (SumeetBatra) [reply gym_art/quadrotor_multi/quadrotor_multi.py:293]: I've set the method's 'coeff; param default value to be np.inf which sets the penalty to be all zeros.

**Review by alex-petrenko — APPROVED** [review]

**Comment by alex-petrenko** [comment]
@SumeetBatra or @tushartk can you resolve the collisions please? You are better qualified to do this than I am :)
Don't forget to test after the merge

**Pull request merged**